From: =?utf-8?q?Christian_G=C3=B6ttsche?= <cgzones@googlemail.com>
Date: Wed, 30 Aug 2017 18:35:09 +0200
Subject: logrotate.service: add hardening options

---
 examples/logrotate.service | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/examples/logrotate.service b/examples/logrotate.service
index e276878..19b0a34 100644
--- a/examples/logrotate.service
+++ b/examples/logrotate.service
@@ -6,6 +6,18 @@ ConditionACPower=true
 [Service]
 Type=oneshot
 ExecStart=/usr/sbin/logrotate /etc/logrotate.conf
+
+# performance options
 Nice=19
 IOSchedulingClass=best-effort
 IOSchedulingPriority=7
+
+# security options
+ProtectSystem=full
+ProtectHome=true
+PrivateDevices=true
+# breaks SELinux support
+#ProtectKernelTunables=true
+ProtectControlGroups=true
+ProtectKernelModules=true
+PrivateNetwork=true
