From: =?utf-8?q?Christian_G=C3=B6ttsche?= <cgzones@googlemail.com>
Date: Tue, 19 Sep 2017 08:18:47 +0200
Subject: logrotate.c: add compile time configs to version information

---
 logrotate.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/logrotate.c b/logrotate.c
index 4ad58d4..846c830 100644
--- a/logrotate.c
+++ b/logrotate.c
@@ -2600,6 +2600,22 @@ int main(int argc, const char **argv)
 		break;
 	case 'V':
 	    fprintf(stderr, "logrotate %s\n", VERSION);
+	    fprintf(stderr, "\n");
+	    fprintf(stderr, "    Default mail command: %s\n", DEFAULT_MAIL_COMMAND);
+	    fprintf(stderr, "    Compress command:     %s\n", COMPRESS_COMMAND);
+	    fprintf(stderr, "    Uncompress command:   %s\n", UNCOMPRESS_COMMAND);
+	    fprintf(stderr, "    Compress extension:   %s\n", COMPRESS_EXT);
+	    fprintf(stderr, "    State file path:      %s\n", STATEFILE);
+#ifdef WITH_ACL
+	    fprintf(stderr, "    ACL support:          yes\n");
+#else
+	    fprintf(stderr, "    ACL support:          no\n");
+#endif
+#ifdef WITH_SELINUX
+	    fprintf(stderr, "    SELinux support:      yes\n");
+#else
+	    fprintf(stderr, "    SELinux support:      no\n");
+#endif
 	    poptFreeContext(optCon);
 	    exit(0);
 	default:
