From: Paul Martin <pm@debian.org>
Date: Wed, 30 Aug 2017 16:24:03 +0200
Subject: chown-484762

===================================================================
---
 logrotate.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/logrotate.c b/logrotate.c
index 4ad58d4..752a9ef 100644
--- a/logrotate.c
+++ b/logrotate.c
@@ -510,8 +510,12 @@ static int createOutputFile(char *fileName, int flags, struct stat *sb,
     if (fchmod(fd, (S_IRUSR | S_IWUSR) & sb->st_mode)) {
 	message(MESS_ERROR, "error setting mode of %s: %s\n",
 		fileName, strerror(errno));
-	close(fd);
-	return -1;
+	if (geteuid() == 0) {
+		/* If run as non-root, we may not be able to duplicate
+		   the ownership of the file. Warn but don't fail. */
+		close(fd);
+		return -1;
+	}
     }
 
 	if (fstat(fd, &sb_create)) {
