From: Paul Martin <pm@debian.org>
Date: Wed, 30 Aug 2017 16:24:03 +0200
Subject: taboo-716649

===================================================================
---
 config.c    | 2 ++
 logrotate.8 | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/config.c b/config.c
index 037d593..c973bc3 100644
--- a/config.c
+++ b/config.c
@@ -130,7 +130,9 @@ static const char *defTabooExts[] = {
 	",v",
 	".cfsaved",
 	".disabled",
+	".dpkg-bak",
 	".dpkg-dist",
+	".dpkg-del",
 	".dpkg-new",
 	".dpkg-old",
 	".rhn-cfg-tmp-*",
diff --git a/logrotate.8 b/logrotate.8
index dd22344..4503a17 100644
--- a/logrotate.8
+++ b/logrotate.8
@@ -576,7 +576,9 @@ is replaced. At startup, the taboo extension list
 .IR ,v ,
 .IR .cfsaved ,
 .IR .disabled ,
+.IR .dpkg\-bak ,
 .IR .dpkg\-dist ,
+.IR .dpkg\-del ,
 .IR .dpkg\-new ,
 .IR .dpkg\-old ,
 .IR .rhn\-cfg\-tmp\-* ,
